Vending Goods

BloodyMan opened the pull request:
:cl: Cajoes
rscadd: food in steel cans and support for canned items, added to cargo
rscadd: Added pistachios to snix vendor
rscadd: Added Sol Snacks vending machine. (And ethnic sol foods to go with it) - Vendor by Chinsky
rscadd: Added japenese snacks to the snix machine and changed it to its own vendor with some label tweaks please welcome the painfully ethnic "Yummy Fods" vending machine. Available in cargo
rsctweak: Snix machine now more slavic in design
rscadd: Added deoderant to lavatory essentials
rscadd: Added chessboard to game vendor
rscadd: Added numerous vendors to the Cargo Supply Form
/:cl:

MistakeNot4892 commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -3071,8 +3071,116 @@
+/obj/item/weapon/reagent_containers/food/snacks/canned/attack_self(var/mob/user)
+	if(sealed)
+		playsound(loc,'sound/effects/canopen.ogg', rand(10,50), 1)
+		to_chat(user, "<span class='notice'>You unseal \the [src] with an crack of metal.</span>")
```
`an crack`

BlueNexus commented:
Might want to make it so you can open them with the Combi-knife can opener.

BloodyMan commented:
> Might want to make it so you can open them with the Combi-knife can opener.

They have observable pull-ring technology for ease of opening. - But if I can work out how that bit of code works, I will consider it for future updates. Just to piss people off a bit.

CakeQ commented:
You need to replace instances of caviar in away site maps with the correct path

## TESTING: Spawning Vox Base
```
## ERROR: Couldn't find atom path specified in map: /obj/item/weapon/reagent_containers/food/snacks/caviar
## ERROR: Couldn't find atom path specified in map: /obj/item/weapon/reagent_containers/food/snacks/caviar
## ERROR: Couldn't find atom path specified in map: /obj/item/weapon/reagent_containers/food/snacks/caviar
## ERROR: Couldn't find atom path specified in map: /obj/item/weapon/reagent_containers/food/snacks/caviar
```

comma commented:
Also conflicts with your monkeywrap stuff

BloodyMan committed:
Pushing Ahead

icons for solvend

vendor-prices

commas

vend

qualifier

done

dveg

request fixes

completion

bread fix

Vital fixes

snixmachinereborn

icon-conflict-resolution

fixes

comma commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -3475,6 +3586,124 @@
+	icon_state = "mochicake_wrapped"
+	desc = "Konnichiwa!"
+	trash = /obj/item/trash/mochicakewrap
+	filling_color = "#ffffff"
```
don't need to redefine all the variables that are same as on parent
```
	filling_color = "#ffffff"
	center_of_mass = "x=15;y=9"
	nutriment_amt = 5
	bitesize = 2
```

comma commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -3475,6 +3586,124 @@
+	icon_state = "mooncake_wrapped"
+	desc = "Explore the dark side! May contain trace amounts of reconstituted cocoa."
+	trash = /obj/item/trash/mooncakewrap
+	filling_color = "#ffffff"
```
filling_color is this value by default for all food items, you do not need to overridew it /every/ /single/ /time/

comma commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -3073,8 +3073,128 @@
+		to_chat(user, "<span class='notice'>You unseal \the [src] with a crack of metal.</span>")
+		unseal()
+
+/obj/item/weapon/reagent_containers/food/snacks/canned/on_update_icon()
```
Not seeing where actual icon update is called.

BloodyMan replied:
Every time you open a canned food.

BloodyMan replied:
Change denied.

comma commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -3073,8 +3073,128 @@
+
+/obj/item/weapon/reagent_containers/food/snacks/canned/Initialize()
+	. = ..()
+	atom_flags ^= ATOM_FLAG_OPEN_CONTAINER
```
Doesn't need to be set in initalize, just set variable to 0 in type define.

BloodyMan replied:
Bad command and or file name

comma replied:
```
/obj/item/weapon/reagent_containers/food/snacks/canned
	name = "void can"
	icon = 'icons/obj/food_canned.dmi'
        atom_flags = 0
	var/sealed = TRUE
```

BloodyMan replied:
command parsed

comma commented:
Not seeing directional sprites for that vendomat

BloodyMan commented:
> Not seeing directional sprites for that vendomat

Work in progress,  -  Any suggestions?

BloodyMan commented:
This PR is probably as finished as its gonna get.

BloodyMan commented:
Merge conflict resolved, - ready for review
